### DISCUSSION TITLE
Best way to check results for each test?

### DISCUSSION THREAD

**Discussion by ShawnTheBeachy** [thomhurst/TUnit#1252] [Q&A] [answered]
Based on [these](https://thomhurst.github.io/TUnit/docs/tutorial-extras/cleanup) docs and [these](https://thomhurst.github.io/TUnit/docs/tutorial-extras/test-context) docs, I'm trying to run a static method after every test to save a trace if the test failed. From the docs, it sounds to me like the test results should be available at this point. However, they're still null.

```c#
[AfterEvery(Test)]
public static async Task SaveTraceIfFailed(TestContext testContext)
{
    ...
}
```

Is there a better place to be doing something like this?

**Comment by thomhurst** [comment, marked as answer]
> the test results should be available at this point. However, they're still null

This is a bug, I'll get a fix out for you later today

**Reply by ShawnTheBeachy** [reply]
Awesome, thank you! That makes me wonder if this other issue I was hitting is also a bug, rather than me just not understanding it... I've opened a separate discussion [here](https://github.com/thomhurst/TUnit/discussions/1253)

**Reply by thomhurst** [reply]
Should be resolved in 0.3.34
